src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -178,11 +217,12 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
   const hasSavedEvents = schedule.personalEvents.length > 0;
 
   const downloadPersonalEventsCalendar = useCallback(() => {
-    const dayStart = addDays(startOfToday(), selectedDayIndex);
+    const startOfSelectedDay = addDays(firstDayOfSchedule, selectedDayIndex);
+    // @debt this seems to only download events for the currently selected day, yet the UI implies as though it will download all of my events
     const allPersonalEvents: PersonalizedVenueEvent[] = relatedVenueEvents
       .map(
         prepareForSchedule({
-          day: dayStart,
+          day: startOfSelectedDay,
           usersEvents: userEventIds,
           isForCalendarFile: true,
         })
```

Comment:
[debt] this seems to only download events for the currently selected day, yet the UI implies as though it will download all of my events